website/server/models/group.js

```diff
@@ -867,6 +868,10 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
 
   // If user is the last one in group and group is private, delete it
   if (group.memberCount <= 1 && group.privacy === 'private') {
+    let plan = group.purchased.plan;
```

Comment:
this should be moved up in the code because some actions have already happened like await `Bluebird.all(challengesToRemoveUserFrom);` and `User.update`. It's possible some has already happened in the controller that calls group.leave, in that case the check should be moved there